### PR TITLE
[Tests] log when openssl doesn't support cipher 

### LINKED ISSUE #37: test failures depending on openssl version/flags
Typically, on debian openssl has a stricter set of algorithms and hashes.
Some fixtures will make the test fail, like ec.dsa.
It might be a good idea to not test them, or to catch the error and not fail here:
https://github.com/crypto-browserify/browserify-sign/blob/44a10f6edf5df9dade14cf5600392417da6f966b/test/index.js#L66

### PR DESCRIPTION
**Pull request by kapouer**
This fixes #37.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -63,8 +63,15 @@ fixtures.valid.ec.forEach(function (f) {
   }
 
   test(f.message, function (t) {
-    var nSign = nCrypto.createSign(f.scheme)
-    var bSign = bCrypto.createSign(f.scheme)
+    var nSign, bSign
+    try {
+      nSign = nCrypto.createSign(f.scheme)
```
Comment (calvinmetcalf) [inline test/index.js]: only this one needs to be in the try catch

File: test/index.js [hunk 2]
```diff
@@ -63,8 +63,15 @@ fixtures.valid.ec.forEach(function (f) {
...
+      bSign = bCrypto.createSign(f.scheme)
+    } catch(ex) {
```
Comment (calvinmetcalf) [inline test/index.js]: need a space after the catch

File: test/index.js [hunk 3]
```diff
@@ -63,8 +63,15 @@ fixtures.valid.ec.forEach(function (f) {
...
+      console.info("skipping unsupported scheme", f.scheme)
```
Comment (calvinmetcalf) [inline test/index.js]: need single quotes

**Review by calvinmetcalf — DISMISSED** [review]
looks good except for the standard nits, also we probably want to to put a try catch around the stuff on line 30 too.  Thanks for doing this.

**Comment by kapouer** [comment]
Seeing the fix for crypto-browserify i suppose that workaround is not right, then.

**Comment by calvinmetcalf** [comment]
well the main one was just testing a single type of signature, so I just changed which one it was, for this one we have to test ALL of them so I think this is the right way.

File: test/index.js (hunk 1 above)
Comment (kapouer) [reply test/index.js]: moved outside

File: test/index.js (hunk 2 above)
Comment (kapouer) [reply test/index.js]: added

File: test/index.js (hunk 3 above)
Comment (kapouer) [reply test/index.js]: changed

**Comment by ljharb** [comment]
I rebased this, and will do so again once I get CI restored, and then land it assuming it passes.

**Review by ljharb — APPROVED** [review]

**Pull request merged**